server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -177,8 +185,14 @@ public class TransportSearchAction extends HandledTransportAction<SearchRequest,
 
     private volatile boolean isRequestStatsEnabled;
 
+    private volatile boolean searchQueryCategorizationEnabled;
```

Comment:
This needs to be called as MetricsEnabled matching with the flag.

```suggestion
    private volatile boolean searchQueryMetricsEnabled;
```